sdk/python/toolchain/pip.go

```diff
@@ -26,13 +28,125 @@ import (
 	"strings"
 
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
 )
 
 const (
 	windows             = "windows"
 	pythonShimCmdFormat = "pulumi-%s-shim.cmd"
 )
 
+type pip struct {
+	// The absolute path to the virtual env. Empty if not using a virtual env.
+	virtualenvPath string
+	// The virtual option as set in Pulumi.yaml.
+	virtualenvOption string
+	// The root directory of the project.
+	root string
+}
+
+var _ Toolchain = &pip{}
+
+func newPip(root, virtualenv string) (*pip, error) {
+	virtualenvPath := resolveVirtualEnvironmentPath(root, virtualenv)
+	logging.V(9).Infof("Python toolchain: using pip at %s", virtualenvPath)
```

Comment:
This is very much besides the PR, and I feel like something I should know already, but do we have any docs on log levels anywhere?  I.e. what should be logged at 9 vs. 5 vs. whatever?

Reply:
Yeah I don't know 🤷